### PR TITLE
improve get_mac_address

### PR DESCRIPTION
**Pull request by Stephen1993**
# Description
Please add an informative description that covers that changes made by the pull request and link all relevant issues.
Due to the inability to fix the name of the first network card, for example, for laptops and Mac mini, their names are different, so change from obtaining the first network card MAC address to obtaining all MAC addresses.
image

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/_utils.py [hunk 1]
```diff
@@ -1120,37 +1120,23 @@ def pd_read_json(file) -> "DataFrame":
...
-                mac_address = str(uuid.UUID(int=node).hex[-12:])
-
-        return mac_address
+        mac_address = []
+        net_addresses = psutil.net_if_addrs()
+        # Obtain all MAC addresses, then sort and concatenate them
+        for net_address in net_addresses.values():
+            for net_interface in net_address:
+                if net_interface.family == psutil.AF_LINK and net_interface.address != "00-00-00-00-00-00":
+                    mac_address.append(net_interface.address)
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_utils.py:1134]: where is the sort?

Comment (Stephen1993) [reply src/promptflow/promptflow/_sdk/_utils.py:1134]: `psutil.net_if_addrs()` returns sorted values, but I can sort it again outside.
image

Comment (wangchao1230) [reply src/promptflow/promptflow/_sdk/_utils.py:1134]: looks like it only sort by family, we should sort by address.

Comment (Stephen1993) [reply src/promptflow/promptflow/_sdk/_utils.py:1135]: ok, sorting by address is also acceptable. I just added sorting by address name.

**Review by wangchao1230 — APPROVED** [review]

**Review by crazygao — APPROVED** [review]

**Pull request merged**